mono/mini/method-to-ir.c

```diff
@@ -2160,6 +2160,12 @@ check_method_sharing (MonoCompile *cfg, MonoMethod *cmethod, gboolean *out_pass_
 		*out_pass_mrgctx = pass_mrgctx;
 }
 
+inline static gboolean
+mono_need_unbox_trampoline (MonoMethod *method)
+{
+	return method->klass == mono_defaults.object_class || mono_class_is_interface (method->klass);
```

Comment:
This hopefully isn't all needed -- unable to tailcall an interface would be bad.
What is the unbox trampoline? I see it used only in x86.

Reply:
The unbox trampoline is present on all arches, it's used when you do an interface invoke into a boxed value type.
A boxed value type is represented in memory all of its fields after an object header.

ValueType methods are compiled with `this` as a byref to the first field.
interface invokes are compiled with `this` as a pointer to the object header.

This means we need an unbox trampoline that essentially does `this + sizeof (MonoObject)`.

Reply:
This was just bad guessing on my part. It wasn't so much the questionable changing of tail to false in a few places, such as for the unbox trampoline and llvmonly, which is still questionable, but too much turning into tail in the change. I didn't debug it just a little trial/error based on the original change.

Reply:
Only x86 seems to reference this though.